src/mscorlib/shared/System/Runtime/CompilerServices/ConfiguredValueTaskAwaitable.cs

```diff
@@ -74,7 +74,11 @@ internal ConfiguredValueTaskAwaiter(ValueTask<TResult> value, bool continueOnCap
 
             /// <summary>Gets the task underlying the incomplete <see cref="_value"/>.</summary>
             /// <remarks>This method is used when awaiting and IsCompleted returned false; thus we expect the value task to be wrapping a non-null task.</remarks>
-            (Task task, bool continueOnCapturedContext) IConfiguredValueTaskAwaiter.GetTask() => (_value.AsTaskExpectNonNull(), _continueOnCapturedContext);
```

Comment:
I like the change to not use ValueTuple here. It will improve .NET Core as well.

ValueTuples are pretty expensive types with large footprint, unfortunately. Not as bad as Linq, but bad enough.

Reply:
This makes me sad